### DISCUSSION TITLE
How To Cancel Requests In useFetch?

### DISCUSSION THREAD

**Discussion by flapjalc** [nuxt/nuxt#24257] [Questions] [answered]
In axios I can call the isCancel method. In useFetch, unfortunately, I don't see any way to pass the signal parameter like:

```js
axios.get('https://your-server.com/products', {
  // specify the AbortController signal to use
  // for cancelling the request
  signal: abortController.signal
}).catch(function (e) {
  // if the reason behind the failure 
  // is a cancellation
  if (axios.isCancel(e)) {
    console.error('Operation canceled');
  } else {
    // handle HTTP error...
  }
})
```

**Comment by TheAlexLichter** [comment, marked as answer]
You can use either a custom `$fetch` option or `useAsyncData` + `$fetch`. Either way, `$fetch` takes a `signal` option for an abort controller.

**Reply by MickL** [reply]
Would be amazing if useFetch() provides an abort() function tho

**Reply by TheAlexLichter** [reply]
@MickL PR/Issue welcome!

**Reply by JoJk0** [reply]
It feels like `abort` should be an option by default. I'm migrating app to Nuxt from @antfu 's template and VueUse's `useFetch` function has it implemented.

**Reply by TheAlexLichter** [reply]
@JoJk0 See the comment above from me which still applies 👍 Would be good to have a dedicated issue to implement that if you feel it is needed ☺️ 

> @MickL PR/Issue welcome!
